### PR TITLE
Initial PR: Motr level DI tests

### PR DESCRIPTION
**Pull request by seagate-sarang-sawant**
# Problem Statement
Initial PR to cater Motr level DI tests which utilize M0* utils and validate data
corruption detection.

# Design
-  NA

# Coding
   Checklist for Author
-  [ ] Coding conventions are followed and code is consistent

# Testing 
  Checklist for Author
-  [ ] New/Affected tests are executed on Latest Build
-  [ ] Attach test execution logs
-  [ ] Collection tested and no collection error introduced (`pytest --local True --collect-only`)

# Review Checklist 
  Checklist for Author
- [ ] JIRA number/GitHub Issue added to PR
- [ ] PR is self reviewed
- [ ] Jira and state/status is updated and JIRA is updated with PR link
- [ ] Check if the description is clear and explained

# Impact Analysis
  Checklist for Author/Reviewer/GateKeeper
-  [ ] If change in any common function, make sure to update all calls and execute all affected tests.

# Documentation
  Checklist for Author
-  [ ] Changes done to ReadMe / WIKI / Confluence page / Quick Start Guide

### CODE REVIEW

File: comptests/motr/test_motr_di.py [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+                              '2', '4', '2', '3', '4', '8', '4', '1024']
+                bsize_list = ['4K', '4K', '4K', '8K', '16K', '64K', '64K', '128K',
+                              '4K', '1M', '1M', '4M', '4M', '4M', '4M', '16M', '1M']
+                layout_ids = ['1', '1', '1', '2', '3', '5', '5', '6', '1',
+                              '9', '9', '11', '11', '11', '11', '13', '9']
+                for b_size, count, layout in zip(bsize_list, count_list, layout_ids):
+                    object_id = str(self.system_random.randint(1, 100)) + ":" + \
+                                str(self.system_random.randint(1, 100))
+                    self.motr_obj.dd_cmd(b_size, count, infile, node)
+                    self.motr_obj.cp_cmd(b_size, count, object_id, layout, infile, node, client_num)
```
Comment (zohebkhann) [inline comptests/motr/test_motr_di.py:103]: How are we corrupting the data here using cp_cmd? Or should we use cp_corrupt_cmd?

File: comptests/motr/test_motr_di.py [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+
+    @pytest.mark.tags("TEST-23036")
+    @pytest.mark.motr_sanity
+    def test_m0cp_m0cat_block_corruption(self):
+        """
+        Corrupt data block using m0cp and reading from object with m0cat should error.
+        """
+        self.m0cp_corrupt_m0cat()
+
+    @pytest.mark.tags("TEST-23036")
```
Comment (zohebkhann) [inline comptests/motr/test_motr_di.py:120]: Why are we executing the same test TEST-23036 twice?

File: comptests/motr/test_motr_di.py [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+    @pytest.mark.motr_sanity
+    def test_m0cp_m0cat_checksum_corruption(self):
```
Comment (zohebkhann) [inline comptests/motr/test_motr_di.py:122]: test_m0cp_m0cat_block corrupt is doing the same thing calling function 'm0cp_corrupt_m0cat'. Why two test functions are needed?

File: comptests/motr/test_motr_di.py (hunk 1 above)
Comment (seagate-sarang-sawant) [reply comptests/motr/test_motr_di.py:102]: @zohebkhann  Per Shankar right now he is going to add a switch and hook the datablock corruption code in CP. For long term I want DC command to be created for this purpose specifically.

File: comptests/motr/test_motr_di.py (hunk 2 above)
Comment (seagate-sarang-sawant) [reply comptests/motr/test_motr_di.py]: Corrected this one. The test id is changed now.

File: comptests/motr/test_motr_di.py (hunk 3 above)
Comment (seagate-sarang-sawant) [reply comptests/motr/test_motr_di.py:120]: Checksum corruption implementation would be different. Rest of the flow is same.

File: libs/motr/layouts.py [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+"""
+
+ARCH = {'CONFIG_X86_64': '1', 'CONFIG_AARCH64': '5'}
+
+bsize_list = ['4K', '8K', '16K', '32K', '64K', '128K', '256K', '512K',
+              '1M', '2M', '4M', '8M', '16M', '32M']
+
+layout_ids = ['1', '2', '3', '4', '5', '6', '7', '8',
+              '9', '10', '11', '12', '13', '14']
+
```
[the remark this answers is not in the document]
Comment (seagate-sarang-sawant) [reply libs/motr/layouts.py:34]: Lets move the BSIZE_LAYOUT_MAP and reverse map to layouts. Init is not intuitive.

[the remark this answers is not in the document]
Comment (zohebkhann) [reply libs/motr/layouts.py:34]: Ok sure..

File: comptests/motr/test_motr_di.py (hunk 3 above)
Comment (zohebkhann) [reply comptests/motr/test_motr_di.py:120]: ok

**Review by zohebkhann — APPROVED** [review]

File: libs/motr/layouts.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+On RHEL aarch64 platform the page size is 64KB, hence the default layout id must be 5.
+Ref motr/Layout.h for details.
+"""
+
+ARCH = {'CONFIG_X86_64': '1', 'CONFIG_AARCH64': '5'}
+
+bsize_list = ['4K', '8K', '16K', '32K', '64K', '128K', '256K', '512K',
+              '1M', '2M', '4M', '8M', '16M', '32M']
+
+layout_ids = ['1', '2', '3', '4', '5', '6', '7', '8',
```
Comment (gaurkrgaurav) [inline libs/motr/layouts.py:32]: do we need layout id now?

Comment (seagate-sarang-sawant) [reply libs/motr/layouts.py:32]: No, it's the same , where do you see another list ?

**Review by gaurkrgaurav — APPROVED** [review]

**Review by ravindrachoudhari — APPROVED** [review]

**Pull request merged**